src/main.cpp

```diff
@@ -2953,7 +3004,7 @@ bool CheckBlock(const CBlock& block, CValidationState& state, bool fCheckPOW, bo
     // because we receive the wrong transactions for it.
 
     // Size limits
-    if (block.vtx.empty() || block.vtx.size() > MAX_BLOCK_SIZE || ::GetSerializeSize(block, SER_NETWORK, PROTOCOL_VERSION) > MAX_BLOCK_SIZE)
+    if (block.vtx.empty() || block.vtx.size() > MAX_BLOCK_SIZE || GetVirtualBlockSize(block) > MAX_BLOCK_SIZE)
```

Comment:
It's possible for someone to malleate a block so that it fails the virtual block size check, even if there exists another block with the same hash which is valid.  So the "corruption possible" field ought to be set to be safe, or we could perhaps move this check to ContextualCheckBlock where we can make a more accurate determination of malleability.

This may well be safe here anyway because ProcessNewBlock currently has a redundant call to CheckBlock that prevents it from marking blocks as permanently failed (that happens in AcceptBlock), though I recently opened a PR upstream to get rid of that call.